### PR TITLE
Convert `oBehParams` to be unsigned, and fix `GLOBAL_STAR_IDS` to support more than 128 stars

### PR DESCRIPTION
**Pull request by gheskett**

### CODE REVIEW

File: src/game/behaviors/spawn_star.inc.c [hunk 1]
```diff
@@ -13,9 +13,9 @@ static struct ObjectHitbox sCollectStarHitbox = {
 };
 
 void bhv_collect_star_init(void) {
-    s8 starId = GET_BPARAM1(o->oBehParams);
+    u8 starId = GET_BPARAM1(o->oBehParams);
 #ifdef GLOBAL_STAR_IDS
-    u8 currentLevelStarFlags = save_file_get_star_flags((gCurrSaveFileNum - 1), COURSE_NUM_TO_INDEX(starId / 7));
+    u8 currentLevelStarFlags = save_file_get_star_flags((gCurrSaveFileNum - 1), COURSE_NUM_TO_INDEX((s32) starId / 7));
```
Comment (arthurtilly) [inline src/game/behaviors/spawn_star.inc.c:18]: unnecesary cast

File: src/game/behaviors/sparkle_spawn_star.inc.c [hunk 2]
```diff
@@ -16,7 +16,7 @@ void bhv_spawned_star_init(void) {
     if (!(o->oInteractionSubtype & INT_SUBTYPE_NO_EXIT)) {
         o->oBehParams = o->parentObj->oBehParams;
     }
-    s32 param = GET_BPARAM1(o->oBehParams);
+    u32 param = GET_BPARAM1(o->oBehParams);
```
Comment (arthurtilly) [inline src/game/behaviors/sparkle_spawn_star.inc.c:19]: not too related but i feel like this should be a u8 for consistency with the other star code, and why is there no GLOBAL_STAR_IDS handling for this?

File: src/game/behaviors/snowman.inc.c [hunk 3]
```diff
@@ -155,7 +155,7 @@ void bhv_snowmans_bottom_loop(void) {
 
 void bhv_snowmans_head_init(void) {
     u8 starFlags = save_file_get_star_flags(gCurrSaveFileNum - 1, COURSE_NUM_TO_INDEX(gCurrCourseNum));
-    s8 behParams = GET_BPARAM1(o->oBehParams);
+    u8 behParams = GET_BPARAM1(o->oBehParams);
```
Comment (arthurtilly) [inline src/game/behaviors/snowman.inc.c:158]: i think there are other objects that need this too, unagi comes to mind, did you make sure to check everything that uses bparam1?

File: src/game/behaviors/rotating_platform.inc.c [hunk 4]
```diff
@@ -32,7 +32,7 @@ void bhv_wf_rotating_wooden_platform_loop(void) {
 }
 
 void bhv_rotating_platform_loop(void) {
-    s8 speed = GET_BPARAM1(o->oBehParams);
+    s8 speed = (s8) GET_BPARAM1(o->oBehParams);
```
Comment (arthurtilly) [inline src/game/behaviors/rotating_platform.inc.c:35]: VERY useless cast

File: src/game/behaviors/goomba.inc.c [hunk 5]
```diff
@@ -285,7 +285,7 @@ static void floomba_act_startup(void) {
     
     cur_obj_unhide();
 
-    if ((GET_BPARAM3(o->oBehParams) & 0x7F) > o->oZoomCounter) {
+    if ((s32) (GET_BPARAM3(o->oBehParams) & 0x7F) > o->oZoomCounter) {
```
Comment (arthurtilly) [inline src/game/behaviors/goomba.inc.c:288]: incredibly useless cast, the & 0x7F would always return an unsigned number anyway

**Review by arthurtilly — CHANGES_REQUESTED** [review]

File: src/game/behaviors/snowman.inc.c (hunk 3 above)
Comment (gheskett) [reply src/game/behaviors/snowman.inc.c:158]: I did a search on everything `BPARAM1` and also a regex search against `oBehParams` and `>> 24`.

Where does unagi even use bparam1? I don't see anything for it in the behavior file.

File: src/game/behaviors/goomba.inc.c (hunk 5 above)
Comment (gheskett) [reply src/game/behaviors/goomba.inc.c:288]: This throws a compiler warning without it, it's either cast that or cast o->oZoomCounter

Comment (arthurtilly) [reply src/game/behaviors/goomba.inc.c:288]: shouldn't oZoomCounter be a u32 anyway?

File: src/game/behaviors/snowman.inc.c (hunk 3 above)
Comment (arthurtilly) [reply src/game/behaviors/snowman.inc.c:158]: it should be fine then, basically any object that spawns a star uses its bparam1 for the star id but i think this is handled in the star spawn function based off gcurrentobject

File: src/game/behaviors/goomba.inc.c (hunk 5 above)
Comment (gheskett) [reply src/game/behaviors/goomba.inc.c:288]: Fair point

File: src/game/behaviors/sparkle_spawn_star.inc.c (hunk 2 above)
Comment (gheskett) [reply src/game/behaviors/sparkle_spawn_star.inc.c:19]: true, nice catch

**Review by arthurtilly — APPROVED** [review]
approved

File: src/game/behaviors/sparkle_spawn_star.inc.c [hunk 6]
```diff
@@ -16,8 +16,13 @@ void bhv_spawned_star_init(void) {
     if (!(o->oInteractionSubtype & INT_SUBTYPE_NO_EXIT)) {
         o->oBehParams = o->parentObj->oBehParams;
     }
-    s32 param = GET_BPARAM1(o->oBehParams);
+#ifdef GLOBAL_STAR_IDS
+    u8 param = GET_BPARAM1(o->oBehParams);
```
Comment (gheskett) [inline src/game/behaviors/sparkle_spawn_star.inc.c:20]: Self note: move this outside of the ifdef

**Pull request merged**